src/shortcuts/__tests__/deleteEmptyThoughtOrOutdent.ts

```diff
@@ -77,3 +78,34 @@ it('do not outdent thought with siblings', () => {
 
   expect(exported).toEqual(expectedOutput)
 })
+
+describe('DOM', () => {
+  beforeEach(async () => {
+    await createTestApp()
+  })
+
+  afterEach(cleanupTestApp)
+
+  it('delete the thought when user triggered clearThought and then hit back', async () => {
+    store.dispatch([
+      newThought({ value: 'a' }),
+      newThought({ value: 'b', insertNewSubthought: true }),
+      setCursor({ path: rankThoughtsFirstMatch(store.getState(), ['a', 'b']) })
+    ])
+
+    const bothThoughtsExported = `- a
+  - b`
+
+    /** This ensures that the thought b exists so we can confirm later that it is deleted. */
+    const initialExportedData = exportContext(store.getState(), ['a'], 'text/plain')
+    expect(initialExportedData).toBe(bothThoughtsExported)
+
+    executeShortcut(clearThoughtShortcut, { store })
+
+    executeShortcut(deleteEmptyThoughtOrOutdent, { store })
+
+    /** This ensures that the thought b doesn't exist now. */
+    const exported = exportContext(store.getState(), ['a'], 'text/plain')
+    expect(exported).toBe(`a`)
```

Comment:
Actually, this ensures that thought `b` is not within the context `a`. If thought `b` is deleted, it is not in the context `a`. Unfortunately, if thought `b` is outdented (the fail case), thought `b` is *also* not in context `a`. It has just been moved to the root. So this doesn't actually fail when the bug is present.

If you export the entire thoughtspace with `exportContext(store.getState(), [HOME_TOKEN], 'text/plain')`, then it will export all thoughts, and you can be sure that `b` has been deleted.

Reply:
I changed the context to `HOME_TOKEN` for both the tests before/after.